packages/idempotency/src/persistence/DynamoDbPersistenceLayer.ts

```diff
@@ -1,18 +1,82 @@
-import { IdempotencyRecord, PersistenceLayer } from './PersistenceLayer';
+import { DynamoDB, DynamoDBServiceException } from '@aws-sdk/client-dynamodb';
+import { DynamoDBDocument, GetCommandOutput } from '@aws-sdk/lib-dynamodb';
+import { IdempotencyItemAlreadyExistsError, IdempotencyItemNotFoundError } from '../Exceptions';
+import { IdempotencyRecordStatus } from '../types/IdempotencyRecordStatus';
+import { PersistenceLayer } from './PersistenceLayer';
+import { IdempotencyRecord } from './IdempotencyRecord';
 
 class DynamoDBPersistenceLayer extends PersistenceLayer {
-  public constructor(_tableName: string, _key_attr: string = 'id') {
+  private _table: DynamoDBDocument | undefined;
+
+  public constructor(private tableName: string, private key_attr: string = 'id',
+    private status_attr: string = 'status', private expiry_attr: string = 'expiration',
+    private in_progress_expiry_attr: string = 'in_progress_expiry_attr',
+    private data_attr: string = 'data') {
     super();
   }
-  protected async _deleteRecord(): Promise<void> {}
-  protected async _getRecord(): Promise<IdempotencyRecord> {
-    return Promise.resolve({} as IdempotencyRecord);
+
+  protected async _deleteRecord(record: IdempotencyRecord): Promise<void> {
+    const table: DynamoDBDocument = this.getTable();
+    await table.delete({
+      TableName: this.tableName, Key: { [this.key_attr]: record.idempotencyKey }
+    });
+  }
+
+  protected async _getRecord(idempotencyKey: string): Promise<IdempotencyRecord> {
+    const table: DynamoDBDocument = this.getTable();
+    const output: GetCommandOutput = await table.get(
+      {
+        TableName: this.tableName, Key: { [this.key_attr]: idempotencyKey }
+      }
+    );
+
+    if (!output.Item) {
+      throw new IdempotencyItemNotFoundError();
+    }
+
+    return new IdempotencyRecord(output.Item[this.key_attr], output.Item[this.status_attr], output.Item[this.expiry_attr], output.Item[this.in_progress_expiry_attr], output.Item[this.data_attr], undefined);
+  }
+
+  protected async _putRecord(_record: IdempotencyRecord): Promise<void> {
+    const table: DynamoDBDocument = this.getTable();
+
+    const item = { [this.key_attr]: _record.idempotencyKey, [this.expiry_attr]: _record.expiryTimestamp, [this.status_attr]: _record.getStatus() };
+
+    const idempotencyKeyDoesNotExist = 'attribute_not_exists(#id)';
+    const idempotencyKeyExpired = '#expiry < :now';
+    const notInProgress = 'NOT #status = :inprogress';
+    const conditionalExpression = `${idempotencyKeyDoesNotExist} OR ${idempotencyKeyExpired} OR ${notInProgress}`;
+    try {
+      await table.put({ TableName: this.tableName, Item: item, ExpressionAttributeNames: { '#id': this.key_attr, '#expiry': this.expiry_attr, '#status': this.status_attr }, ExpressionAttributeValues: { ':now': Date.now() / 1000, ':inprogress': IdempotencyRecordStatus.INPROGRESS }, ConditionExpression: conditionalExpression });
```

Comment:
Do you have any specific recommendations for configurating the linting for this package like the rest of the project? I tried adding this package to the lerna.json so it would run the same lint-fix command on the idempotency package, but it did not fix the issue. Running lint fix on the project itself also doesn't do anything different.

Reply:
@vgphoenixcampos  Embarrassingly, we actually don't have this. I use the default one on VSCode but it doesn't break objects into multiple lines.

@dreamorosi  How do you this?  I'm thinking about using [prettier](https://prettier.io/docs/en/cli.html) as part of `lint-fix` target or commit hook. It has plugins for both [VSCode](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) and [IntelliJ](https://www.jetbrains.com/help/idea/prettier.html). 

Then we will have a massive PR with only formatting :D

Reply:
Yea, let's park this topic for now and prioritise the logic, you can disregard my comment for now.

I like prettier or similar but I'd like to find a way to do it without messing entirely the history. I know there might be ways but it requires some investigation. For now let's just work with `eslint` with the config that is already in the project.